mkdocs/technical/plugins/hooks.md

```diff
@@ -779,5 +780,151 @@ Not all the hooks have been documented yet. ( [help needed! &#128568;](https://d
             - [Plugin object](https://github.com/cheshire-cat-ai/core/blob/main/core/cat/mad_hatter/plugin.py#L25)
 
     3. **Input arguments**  
+        This hook has no input arguments.
+
+        ??? example
+
+            ```python
+            from cat.mad_hatter.decorators import plugin
+            from pydantic import BaseModel, Field
+
+            # define your plugin settings model
+            class MySettings(BaseModel):
+            text_field: str = Field(
+                title="Your field title",
+                description="Your field description",
+                default="""default value""",
+                extra={"type": "Text"}
+            )
+            bool_field: bool = Field(
+            default=True,
+            title="Your field title",
+            )
+            int_field: int = 3
+
+            # get your plugin settings schema
+            @plugin
+            def settings_schema():
+                return MySettings.model_json_schema()
+
+            # load your plugin settings
+            settings = ccat.mad_hatter.get_plugin().load_settings()
+            # access settings
+            text_field = settings["text_field"]
+            bool_field = settings["bool_field"]
+            int_field = settings["int_field"]
+            ```
+
+        ??? warning
+
+            Note that `settings["field_name"]` works if you have a `settings.json` file in your plugin folder,
+            otherwise you get `KeyError`.
+
+            You can solve that by:
+
+            1. creating a default `settings.json` file
+            2. using a `try-catch` block when accessing plugin settings
+
+            Soon, `settings.json` will be created by the cat core for the *settings fields* with default values only.
```

Comment:
Since this is happening in a few days, I would consider it already effective to avoid having to modify the doc again once this feature will be on `main` branch

Reply:
ok np will leave a note about it saying that, a default `settings.json` is created for the settings fields that have default values